### ISSUE TITLE
screen-ls: shebang line is missing

### ISSUE THREAD

**Issue by salewski** [salewski/screen-wrappers#21]
<pre>
$ <strong>screen-ls --version</strong>
screen-ls 0.3.4  (built: 2019-01-26 12:26:56)

Copyright (C) 2013, 2016, 2017, 2019 Alan D. Salewski <salewski@att.net>
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Alan D. Salewski.
</pre>

It is odd that this has not been noticed earlier, but the `screen-ls` script generated from `screen-ls.in` does not have a shebang line. The first handful of lines of the generated program are (with line numbers prefixed):
```
   1 # Generated from screen-ls.in; do not edit by hand.
   2 #
   3 # Copyright (C) 2013, 2016, 2017, 2019 Alan D. Salewski <salewski@att.net>
   4 #
   5 #    This program is free software; you can redistribute it and/or modify
   6 #    it under the terms of the GNU General Public License as published by
```

The first line should be a shebang line that references the `bash` found (or explicitly specified by the user) at configure time. It should look something like this:
```bash
#! /usr/bin/bash -
```

The script happens to work when Bash is the the user's interactive shell, or when invoked from a script that Bash is running. The reason is due to the long-standing Unix tradition of treating text files as shell scripts when their executable bit is set, even if they lack a shebang line. (That behavior predates the invention of the shebang line.) The system returns `ENOEXEC` to the invoking process, and in the case of Bash the shell attempts to interpret the lines of the text file as shell commands. (Other shells do something similar, or explicitly invoke `/bin/sh` with the text file as an argument.)

It was never intended that the `screen-ls` program would depend on the above described behavior.

**Comment by salewski** [comment]
The generated shebang line was removed from `screen-ls` with commit `d3024d90b735ef6e884e7ff50a21d56ae17a0c02`. I can only assume its removal was unintentional.

For easy reference, here's the Git commit message for that change:
```
commit d3024d90b735ef6e884e7ff50a21d56ae17a0c02
Author: Alan D. Salewski <salewski@att.net>
Date:   Mon Jan 21 18:53:38 2019 -0500

    fixes #14: screen-ls: refactored parsing of `screen --version` output
```

**Comment by salewski** [comment]
The issue has been fixed for 'screen-wrappers-0.3.5'

[The blurb about it](../blob/master/NEWS#issue-21-screen-ls-shebang-line-is-missing) from our ['NEWS'](../blob/master/NEWS) file:
```
*** issue #21: screen-ls: shebang line is missing

    https://github.com/salewski/screen-wrappers/issues/21

    At build time, a shebang line should be inserted into the generated
    'screen-ls' program, using the 'bash' executable found at configure
    time. The build used to do so, and has been fixed to do so again.

    This bug was introduced in screen-wrappers 0.3.3 (released 2019-01-21),
    and went undetected for years -- probably because using Bash for one's
    interactive shell is so common; the script happens to work when Bash is
    the the user's interactive shell, or when invoked from a script that Bash
    is running. The reason is due to the long-standing Unix tradition of
    treating text files as shell scripts when their executable bit is set,
    even if they lack a shebang line. (That behavior predates the invention of
    the shebang line.) The system returns ENOEXEC to the invoking process, and
    in the case of Bash the shell attempts to interpret the lines of the text
    file as shell commands. (Other shells do something similar, or explicitly
    invoke /bin/sh with the text file as an argument.)

    It was never intended that the screen-ls program would depend on the above
    described behavior.
```
